### DISCUSSION TITLE
Deploy History: Git SHAs logged anywhere?

### DISCUSSION THREAD

**Discussion by beporter** [dokku/dokku#5114] [Q&A] [answered]
Is there a log file or dokku command I can run that lists the git SHAs that have been used for each deploy? How are folks auditing which commits got `git push`'ed, and in what order? Heroku itself has an activity log at `https://dashboard.heroku.com/apps/$PROJECT_NAME/activity` that provides this info for example.

I've searched the docs, the github issues and these discussions and came up blank. If this isn't already logged, is it something that I could customize in the receive hook myself fairly easily? Or could it get added to the dokku core for git-based deploys?

`dokku git:report` shows the _current_ SHA, but this doesn't help me if I need to look back in the history.

It also doesn't seem to show up in `dokku events`, at least not that I can see.

**Comment by josegonzalez** [comment, marked as answer]
This isn't tracked by Dokku at the moment. I'd considered placing it as part of a builds plugin, but my work on that MR has stalled.

You could probably implement a plugin that stores the information for you wherever you want (offsite in another system like Pagerduty, Fire Hydrant, Sentry, etc.).

**Reply by beporter** [reply]
A local log file would honestly be fine in this case. Even the existing events log noting the SHA attempted/deployed would work!

I've looked through the [dokku git docs](https://dokku.com/docs/deployment/methods/git/) and the [plugin triggers docs](https://dokku.com/docs~v0.20.4/development/plugin-triggers/), but as you said-- the path forward means writing my own entire plugin. Seems like the minimal required effort for that is somewhat above what I was hoping for (dropping a bash script in the right place).

Thanks for the answer! Guess I'll put this on pause for the time being. 👍 

For my own recollection: These might be leads for later:

- https://github.com/fteychene/dokku-build-hook
- https://github.com/mlomnicki/dokku-deploy-hooks

**Reply by josegonzalez** [reply]
You could probably implement a `post-extract` trigger that logs the info you want. The trigger takes the following arguments (in order):

- app name
- the temporary working directory (no .git dir here)
- the commit revision being deployed

Something like this probably works

```
#!/usr/bin/env bash
set -eo pipefail; [[ $DOKKU_TRACE ]] && set -x
source "$PLUGIN_CORE_AVAILABLE_PATH/common/functions"
APP="$1"
TMP_WORK_DIR="$2"
REV="$3" # optional, may not be sent for tar-based builds

logger -f /var/log/dokku/deploy.log -t "deploy.$APP" "$REV"
```
